text_collector_examples/smartmon.sh

```diff
@@ -7,6 +7,10 @@
 #       data in them than you'd think.
 #       http://arstechnica.com/civis/viewtopic.php?p=22062211
 
+PATH="/usr/sbin:$PATH"
+
+set -eu
```

Comment:
While `set -u` is fine, I prefer to not `set -e` and use explicit error handling in bash.

How about `set -u -o pipefail`?

Reply:
@SuperQ What's wrong with -e? I always use -euo pipefail. You can still explicitly handle errors but if you don't, I prefer to fail instead of just continue.

Reply:
`-e` causes the whole script to exit silently in the event of a script failure. I much prefer to capture errors explicitly so I can log the failure.

Reply:
It doesn't cause it to exit silently. What's more, if your systemd job uses `set -o pipefail` due to the pipe to sponge, it causes the job to fail:

```[grahamc@ogden:~]$ systemctl status prometheus-smartmon-exporter.service | cat 
● prometheus-smartmon-exporter.service
   Loaded: loaded (/nix/store/6r5y9vaskll9w8a7zknf2ixnb14bmn5z-unit-prometheus-smartmon-exporter.service/prometheus-smartmon-exporter.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sat 2018-08-25 07:37:09 EDT; 6s ago
  Process: 23351 ExecStart=/nix/store/2amzp591lk45y4ij56z02bcnl8prb0vk-unit-script/bin/prometheus-smartmon-exporter-start (code=exited, status=127)
 Main PID: 23351 (code=exited, status=127)

Aug 25 07:37:09 ogden systemd[1]: Starting prometheus-smartmon-exporter.service...
Aug 25 07:37:09 ogden prometheus-smartmon-exporter-start[23351]: /nix/store/x6fgrwysrldvlg1nnpdqx8gar0fj9jj1-smartmon.sh: line 154: smartctl: command not found
Aug 25 07:37:09 ogden systemd[1]: prometheus-smartmon-exporter.service: Main process exited, code=exited, status=127/n/a
Aug 25 07:37:09 ogden systemd[1]: prometheus-smartmon-exporter.service: Failed with result 'exit-code'.
Aug 25 07:37:09 ogden systemd[1]: Failed to start prometheus-smartmon-exporter.service.
```

I just pushed the `-o pipefail` improvement (thank you,) but I won't remove the `-e` from this change. If someone else would like to send this PR without the `-e`, that would be fine.

Reply:
@grahamc See the above example, bash will exit silently for any command error if you `set -e`, this is why I highly recommend not using it.